src/summary.rs

```diff
@@ -0,0 +1,35 @@
+use std::ffi::{CStr, CString};
+
+use prost::Message;
+
+use crate::bindings::*;
+use crate::error::*;
+use crate::protobuf;
+use crate::summary_result::SummaryResult;
+
+/// Parses the given SQL statement and provides a summary of it.
+///
+/// # Example
+///
+/// ```rust
+/// use pg_query::{Node, NodeEnum, NodeRef};
+///
+/// let result = pg_query::summary("SELECT * FROM contacts", 0, -1);
+/// assert!(result.is_ok());
+/// let result = result.unwrap();
+/// assert_eq!(result.tables(), vec!["contacts"]);
+/// ```
+pub fn summary(statement: &str, parser_options: i32, truncate_limit: i32) -> Result<SummaryResult> {
```

Comment:
It would be good to add documentation for the values that `parser_options` accepts, and what -1 and 0 means when passed to `truncate_limit`

Reply:
It seems like `pg_query::parse()` doesn't actually accept arguments in any way, and I don't see an equivalent that does.

The `pg_query::parse()` signature is:

```rust
pub fn parse(statement: &str) -> Result<ParseResult>
```

And it looks like the function it uses is `pg_query_parse_protobuf(const char *input)`, which calls `pg_query_parse_protobuf_opts(input, PG_QUERY_PARSE_DEFAULT)`.